File: day-1/part_1.rb
INPUT_FILE = "input_1.txt"

def solve()
    max_calories = 0
    current_calories = 0
    f = File.open(INPUT_FILE, "r") do |f|
        f.each { |line|
            line.chomp!
            if line == ""
                if current_calories >= max_calories
                    max_calories = current_calories
                end
                current_calories = 0
            else
                current_calories += Integer(line)
            end
        }    
    end
    puts max_calories
end

solve

File: day-1/part_1.py
INPUT_FILE = "input_1.txt"


def solve():
    max_calories = 0
    current_calories = 0
    with open(INPUT_FILE, "r") as input_file:
        for row in input_file:
            row = row.strip("\n")
            if row == "":
                if current_calories >= max_calories:
                    max_calories = current_calories
                current_calories = 0
            else:
                current_calories += int(row)
    print(max_calories)


if __name__ == "__main__":
    solve()
